docs/development_testing.md

```diff
@@ -7,14 +7,28 @@ As it states in the [Style guidelines section](development_guidelines.md) all co
 - All the unit tests pass
 - All code passes the checks from the linting tools
 
-Local testing is done using Tox, which has been installed as part of running `script/setup` in the [virtual environment](development_environment.md). To start the tests, activate the virtual environment and simply run the command:
+Local testing is done using Tox, which has been installed as part of running `script/setup` in the [virtual environment](development_environment.md). To start the tests, activate the virtual environment (just as you would to run hass) and simply run the command:
 
 ```bash
 $ tox
 ```
+A complete run of every test can take a couple of hours on a beefy PC and need 5GB of disk space -- you probably don't want to do this on your Raspberry Pi.  If successful, Tox will output something like:
```

Comment:
This is not correct. 

The first time takes longer as it needs to build the environment.

Most people will only have 1 Python installed and so it will take shorter (as the others are skipped)

If it all really takes this long, we should start recommending people to just use `py.test tests/components/their_integration`. That will be just seconds.

Reply:
Ok, I decided to test this again to be sure.

I'm running an up-to-date Ubuntu image, nothing other than the default Python installs, inside a VMware player running with 8GB RAM, 8 CPUs, on a M.2 flash drive.  The underlying system is a Intel Core i7-6600K with 32GB of RAM.

I wrote a script which starts with a clean home directory, checks out a clean git repo from github, sets up the venv, and runs hass to make sure it is all working.  It then runs Tox twice.  Once to see the initial test run time, and then a second time immediately afterwards to see the "warm cache" time.

As far as I can tell, I'm following the instructions on the website to the letter.

The first Tox invocation took 1h23m30s.  The second Tox invocation took 1h20m30s.  So building the environment takes roughly 3 minutes.  The resulting disk space consumed was 4970312kB.  The script I ran [can be found here.](https://drive.google.com/open?id=1XBLtbQWrR5jAHLwZRvqzIoQ0OAuzxUSt)

Am I invoking it incorrectly?  Is my VM unusually configured (running stock Ubuntu) or unusually slow for doing Home Assistant development?  I'm trying to update the docs so that the next new developer doesn't spend half a day trying to figure out what they are doing wrong in trying to get the tests to run to submit a simple (<20 line) patch to Home Assistant.  So if you can help me understand what I'm doing wrong, I can perhaps come up with a better patch to the docs.  ;-)

Thanks!

Reply:
I wouldn't describe 1h20m as a couple of hours. 

You are invoking it correctly. The described approach is the approach where as little as possible can go wrong for new users. We might want to document that running `pytest path/to/file.py` directly can be used as an alternative.

Reply:
I think we should recommend `script/lazytox.py` instead. I use it on every one of my PRs. It's fast and it works.

Reply:
Agreed, 1h20m is faster than a couple of hours.  The first time I got this running it did take >2hours, and I'm not entirely sure what I've changed since then.  (I think maybe I increased the resources available to my VM.)